### ISSUE TITLE
Sometimes Failing Test

### ISSUE THREAD

**Issue by maennchen** [c-rack/quantum-elixir#227]
The tests on the CI fail sometimes.

# 1 (Fixed with #228)

```
  1) test add_job/2 adding a unnamed job at run time (Quantum.SchedulerTest)
     test/quantum/scheduler_test.exs:124
     ** (EXIT from #PID<0.335.0>) shutdown: failed to start child: Task.Supervisor
         ** (EXIT) already started: #PID<0.329.0>
```

# 2
```
  1) test skip for current node (Quantum.SchedulerTest)
     test/quantum/scheduler_test.exs:264
     Assertion with == failed
     code:  Quantum.Runner.handle_info(:tick, state1) == {:noreply, state2}
     left:  {:noreply, %{jobs: [nil: %Quantum.Job{name: nil, overlap: true, pids: [], run_strategy: %Quantum.RunStrategy.Random{nodes: [:remote@node]}, schedule: ~e[* * * * * *], state: :active, task: #Function<36.38270216/0 in Quantum.SchedulerTest.test skip for current node/1>, timezone: :utc}], opts: [task_supervisor: Quantum.SchedulerTest.Scheduler.Task.Supervisor, runner: Quantum.SchedulerTest.Scheduler.Runner, global: false, cron: [], timeout: 5000, schedule: nil, overlap: true, timezone: :utc, run_strategy: {Quantum.RunStrategy.Random, :cluster}, jobs: [], otp_app: :scheduler_test, quantum: Quantum.SchedulerTest.Scheduler], reboot: false, date: %NaiveDateTime{calendar: Calendar.ISO, day: 13, hour: 18, microsecond: {0, 0}, minute: 53, month: 7, year: 2017, second: 2}}}
     right: {:noreply, %{jobs: [nil: %Quantum.Job{name: nil, overlap: true, pids: [], run_strategy: %Quantum.RunStrategy.Random{nodes: [:remote@node]}, schedule: ~e[* * * * * *], state: :active, task: #Function<36.38270216/0 in Quantum.SchedulerTest.test skip for current node/1>, timezone: :utc}], opts: [task_supervisor: Quantum.SchedulerTest.Scheduler.Task.Supervisor, runner: Quantum.SchedulerTest.Scheduler.Runner, global: false, cron: [], timeout: 5000, schedule: nil, overlap: true, timezone: :utc, run_strategy: {Quantum.RunStrategy.Random, :cluster}, jobs: [], otp_app: :scheduler_test, quantum: Quantum.SchedulerTest.Scheduler], reboot: false, date: %NaiveDateTime{calendar: Calendar.ISO, day: 13, hour: 18, microsecond: {0, 0}, minute: 53, month: 7, year: 2017, second: 1}}}
     stacktrace:
       (ex_unit) lib/ex_unit/capture_log.ex:77: ExUnit.CaptureLog.capture_log/2
       test/quantum/scheduler_test.exs:283: (test)
```

**Comment by maennchen** [comment]
@c-rack This is only partially resolved.

**Comment by maennchen** [comment]
This issue is potentially solved by #230

**Comment by maennchen** [comment]
Solved with #230
